text/0000-command-overrides.md

```diff
@@ -0,0 +1,149 @@
+
+
+| OCI/Docker   | Kubernetes | Proposed  | Current            |
+| ------------ | ---------- | --------- | ------------------ |
+| `ENTRYPOINT` | `command`  | `command` | `command` + `args` |
+| `CMD`        | `args`     | `args`    | N/A                |
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Let's see how this proposal will work through an example. Let's say we have a process as defined below - 
+
+
+```toml
+[[processes]]
+type = "hi"
+command = ["echo", "Hello"]
+args = ["World"]
+direct = <direct>
+```
+
+If the user runs - 
+
+```
+docker run --entrypoint hi <image-name>
+```
+
+The above would translate to the behavior exhibited by the following process definition as per the current spec -
+
+```toml
+[[processes]]
+type = "hi"
+command = "echo"
+args = ["Hello", "world"]
+direct = <direct>
+```
+
+On the other hand if the user provides arguments by running something like - 
+
+```
+docker run --entrypoint hi <image-name> universe
+```
+
+The above would translate to the behavior exhibited by the following process definition as per the current spec -
+
+```toml
+[[processes]]
+type = "hi"
+command = "echo"
+args = ["Hello", "universe"]
+direct = <direct>
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+This could potentitally cause confusion amongst existing Buildpack authors since the keys in the `launch.toml` remain the same but their meaning and behavior change drastically. However, this would be a one-time change and would make us consistent with other container ecosystems. 
+
+# Alternatives
+[alternatives]: #alternatives
+
+The other alternative to support something like this involves adding an additional `additional_default_args` key in the `Launch Metadata` (launch.toml) that is a list of strings. This list of strings would be a set of additional arguments that would be appended to the existing command invocation if the user does not provide any arguments themselves.
+
+Although this would be a backwards compatible change, this would be an even larger deviation from other container ecosystems and a possible point of confusion for new users in the future.
+
+The above example would look like below if this alternative were to be followed - 
+
+```toml
+[[processes]]
+type = "<process type>"
+command = "<command>"
+args = ["<arguments>"]
+additional_default_args = ["<additional>", "<default>", "<arguments>"]
+direct = <direct>
+```
+
+# Prior Art
+[prior-art]: #prior-art
+
+Discuss prior art, both the good and bad.
+
+- [RFC #45](0045-launcher-arguments.md)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- What parts of the design do you expect to be resolved before this gets merged?
+
+Should we go with the alternative proposal or should we make a backwards incompatible change for a possible improvement in user experience.
```

Comment:
Given that we have a buildpack API which would affect the behaviour along with the fact that we can guide buildpack authors via [migration guides](https://buildpacks.io/docs/reference/spec/migration/) to the new functionality, I don't think this would be as impactful as perceived.

Reply:
Would this also require a bump to the platform API? I'm looking at this line under [launcher](https://github.com/buildpacks/spec/blob/main/platform.md#launcher):
```
MUST append any user-provided <args> to process arguments
```